### ISSUE TITLE
Hamburger menu missing on Safari/macOS/Ventura 13.5.1 (min-width: 768px)

### ISSUE THREAD

**Issue by cazepeda** [statamic/cms#8666]
### Bug description

Hamburger menu is missing on Safari only on desktop view or at (min-width: 768px). The class on `<button aria-label="Toggle Nav" class="nav-toggle hidden md:flex ml-1 shrink-0">...</button>` is using `md:flex`which is causing the hamburger menu to be hidden.

Possible solution is use `md:block` instead. That works cross-browser and does not interfere with mobile view styles.

![missing-hamburger-menu-screenshot-pt1](https://github.com/statamic/cms/assets/24512470/f1a1da45-9a2e-4824-9e37-06438b65aaae)
![missing-hamburger-menu-screenshot-pt2](https://github.com/statamic/cms/assets/24512470/f371129d-e178-449c-959d-0047ade78ac7)
![adjust-nav-toggle-height-to-line-up-hamburger-menu-icon-to-statamic-label](https://github.com/statamic/cms/assets/24512470/9b70689c-ab58-497f-aab6-f2f28e51be19)

### How to reproduce

1. Open Safari browser on macOS (Ventura 13.5.1).
2. Login into control panel.
3. Look for the hamburger menu.

### Environment

```yaml
Environment
Application Name: Statamic
Laravel Version: 10.21.0
PHP Version: 8.2.9
Composer Version: 2.5.8
Environment: local
Debug Mode: ENABLED
URL: testsite.test
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 0
Antlers: runtime
Stache Watcher: Enabled
Static Caching: Disabled
Version: 4.20.0 Solo
```

### Installation

Starter Kit using via CLI

### Antlers Parser

runtime (new)
